src/element-internals.ts

```diff
@@ -136,7 +136,7 @@ export class ElementInternals implements IElementInternals {
    *
    * If the field is valid and a message is specified, the method will throw a TypeError.
    */
-  setValidity(validityChanges: Partial<globalThis.ValidityState>, validationMessage?: string) {
+  setValidity(validityChanges: Partial<globalThis.ValidityState>, validationMessage?: string, anchor?: HTMLElement) {
```

Comment:
Why is this `anchor?: HTMLElement` here?

Reply:
Never mind, I see that this is in the spec, but I'm not sure what it does. Do you have any resources so I can work to make implement that feature? I'm going to approve your PR here.

Reply:
What I know is only from testing on Chrome, I don't have any other resources than the spec, sorry. When I compared how my web components behaved compared to native form elements I noticed that my components did not receive focus like the native ones did on validation errors and I saw this error message in the console:

![image](https://user-images.githubusercontent.com/4290804/107130677-52688580-68d0-11eb-803e-ad2105c0ab06.png)

I noticed the anchor argument in the spec so I tried it out and set it to the native input element inside the shadow dom of my component and voilà: Chrome set focus to that element on validation error and the error was gone.

Reply:
Also found this just now if it helps:

> User agents may focus one of those elements in the process, by running the focusing steps for that element, and may change the scrolling position of the document, or perform some other action that brings the element to the user's attention. For elements that are form-associated custom elements, user agents should use their validation anchor instead, for the purposes of these actions.

 Source: https://html.spec.whatwg.org/multipage/form-control-infrastructure.html